File: src/cb372/intellij/idris/IdrisFileTypeFactory.java
package cb372.intellij.idris;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

public class IdrisFileTypeFactory extends FileTypeFactory {

  @Override
  public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
    fileTypeConsumer.consume(IdrisFileType.INSTANCE, "idr");
  }

}

File: src/main/scala/cb372/intellij/idris/IdrisFileTypeFactory.scala
package cb372.intellij.idris

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

class IdrisFileTypeFactory extends FileTypeFactory {
  def createFileTypes(fileTypeConsumer: FileTypeConsumer) {
    fileTypeConsumer.consume(IdrisFileType.Instance, "idr")
  }
}